lib/kubernetes-deploy/diff_task.rb

```diff
@@ -0,0 +1,315 @@
+# frozen_string_literal: true
+require 'yaml'
+require 'shellwords'
+require 'tempfile'
+require 'fileutils'
+
+require 'kubernetes-deploy/common'
+require 'kubernetes-deploy/concurrency'
+require 'kubernetes-deploy/kubernetes_resource'
+%w(
+  custom_resource
+  cloudsql
+  config_map
+  deployment
+  ingress
+  persistent_volume_claim
+  pod
+  network_policy
+  service
+  pod_template
+  pod_disruption_budget
+  replica_set
+  service_account
+  daemon_set
+  resource_quota
+  stateful_set
+  cron_job
+  job
+  custom_resource_definition
+  horizontal_pod_autoscaler
+  secret
+).each do |subresource|
+  require "kubernetes-deploy/kubernetes_resource/#{subresource}"
+end
+require 'kubernetes-deploy/kubectl'
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/ejson_secret_provisioner'
+require 'kubernetes-deploy/cluster_resource_discovery'
+require 'kubernetes-deploy/template_sets'
+require 'kubernetes-deploy/renderer'
+
+module KubernetesDeploy
+  class DiffTask
+    def server_version
+      kubectl.server_version
+    end
+
+    def initialize(
+      namespace:, context:, current_sha:, logger: nil, kubectl_instance: nil, bindings: {},
+      selector: nil, template_paths: [], template_dir: nil
+    )
+      template_dir = File.expand_path(template_dir) if template_dir
+      template_paths = (template_paths.map { |path| File.expand_path(path) } << template_dir).compact
```

Comment:
You should just be able to use template_paths and drop template_dir. The reason we use both in deploy_task.rb is to ensure the change was backwards compatible at the API-level. Since this is a new task you can just use the newer template_paths (with equivalent -f flag in the CLI)

Reply:
@timothysmith0609 👍 Will do that, what do you think of using "."/current_dir as default in-case of missing though? I did that in Krane version, but i'm not sure if you guys have account for any defaults in newer version, the default for TEMPLATE_DIR was quite a pain tbh :)

Reply:
@timothysmith0609 updated